app/hooks/useAuthorizedSettingsConfig.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  NewDocumentIcon,
+  EmailIcon,
+  ProfileIcon,
+  PadlockIcon,
+  CodeIcon,
+  UserIcon,
+  GroupIcon,
+  LinkIcon,
+  TeamIcon,
+  BeakerIcon,
+  DownloadIcon,
+} from "outline-icons";
+import React from "react";
+import { useTranslation } from "react-i18next";
+import Details from "~/scenes/Settings/Details";
+import Export from "~/scenes/Settings/Export";
+import Features from "~/scenes/Settings/Features";
+import Groups from "~/scenes/Settings/Groups";
+import Import from "~/scenes/Settings/Import";
+import Members from "~/scenes/Settings/Members";
+import Notifications from "~/scenes/Settings/Notifications";
+import Profile from "~/scenes/Settings/Profile";
+import Security from "~/scenes/Settings/Security";
+import Shares from "~/scenes/Settings/Shares";
+import Slack from "~/scenes/Settings/Slack";
+import Tokens from "~/scenes/Settings/Tokens";
+import Zapier from "~/scenes/Settings/Zapier";
+import SlackIcon from "~/components/SlackIcon";
+import ZapierIcon from "~/components/ZapierIcon";
+import env from "~/env";
+import useCurrentTeam from "./useCurrentTeam";
+import usePolicy from "./usePolicy";
+
+type SettingsGroups = "Account" | "Team" | "Integrations";
+type SettingsPage =
+  | "Profile"
+  | "Notifications"
+  | "Api"
+  | "Details"
+  | "Security"
+  | "Features"
+  | "Members"
+  | "Groups"
+  | "Shares"
+  | "Import"
+  | "Export"
+  | "Slack"
+  | "Zapier";
+
+export type ConfigItem = {
+  name: string;
+  path: string;
+  icon: typeof NewDocumentIcon;
```

Comment:
```suggestion
  icon: typeof NewDocumentIcon;
```

This type seems funky